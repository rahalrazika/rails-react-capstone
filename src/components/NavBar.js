import React from 'react';
import user from '../assets/user.png';

function NavBar() {
  return (
    <div className="bg-gray-200 text-white w-64">
      <div>
        <img
          className="inline-block object-cover w-24 h-24 rounded-full  mt-6 mb-4 "
          src={user}
          alt="user"
        />
        <div className="text-gray-500 font-semibold uppercase ">Wesley Larson</div>

      </div>
      <nav>nav</nav>
    </div>
  );
}

export default NavBar;
