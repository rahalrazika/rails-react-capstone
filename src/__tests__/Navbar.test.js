import React from 'react';
import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { MemoryRouter } from 'react-router';
import store from '../redux/store';
import NavBar from '../components/NavBar';

it('renders navbar component correctly', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <MemoryRouter initialEntries={['main']}>
        <NavBar />
      </MemoryRouter>
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
