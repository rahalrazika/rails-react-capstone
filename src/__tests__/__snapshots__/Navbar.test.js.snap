// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders navbar component correctly 1`] = `
<div
  className="flex"
>
  <div
    className="bg-gray-100 text-white w-64 px-2 "
  >
    <div>
      <img
        alt="user"
        className="inline-block object-cover mr-2 bg-gray-100 w-24 h-24 rounded-full  mt-6 mb-4 "
        src={
          Object {
            "canInstrument": true,
            "createTransformer": [Function],
            "getCacheKey": [Function],
            "getCacheKeyAsync": [Function],
            "process": [Function],
            "processAsync": [Function],
          }
        }
      />
      <div
        className="text-gray-500 font-semibold uppercase space-y-6  "
      >
        Wesley Larson
      </div>
    </div>
    <nav>
      <a
        className="block py-2.5 px-4 text-gray-500 font-semibold capitalize rounded transition duration-100  hover:bg-yellow-400 hover:text-white "
        href="/main"
        onClick={[Function]}
      >
        Projects
      </a>
      <a
        className="block py-2.5 px-4 text-gray-500 font-semibold capitalize rounded transition duration-100  hover:bg-yellow-400 hover:text-white "
        href="/fav"
        onClick={[Function]}
      >
        Favourites
      </a>
      <a
        className="block py-2.5 px-4 text-gray-500 font-semibold capitalize rounded transition duration-100  hover:bg-yellow-400 hover:text-white "
        href="/partner"
        onClick={[Function]}
      >
        Partnerships
      </a>
    </nav>
    <div
      className="py-52 text-gray-500 "
    >
      <div
        className="rounded transition duration-100  hover:bg-red-300"
      >
        <button
          className="p-2 uppercase nav-links "
          onClick={[Function]}
          type="button"
        >
          Logout
        </button>
      </div>
      <p>
        help
      </p>
    </div>
  </div>
</div>
`;
